src/pydvl/value/results.py

```diff
@@ -0,0 +1,232 @@
+"""
+This module collects types and methods for the inspection of the results of
+valuation algorithms.
+
+The most important class is :class:`ValuationResult`, which provides access
+to raw values, as well as convenient behaviour as a Sequence with extended
+indexing abilities, and conversion to `pandas DataFrames
+<https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html>`_.
+"""
+import collections.abc
+from dataclasses import dataclass
+from enum import Enum
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Generator,
+    Iterable,
+    List,
+    Optional,
+    Sequence,
+    Union,
+    cast,
+    overload,
+)
+
+import numpy as np
+
+from pydvl.utils import Dataset, SortOrder
+
+try:
+    import pandas  # Try to import here for the benefit of mypy
+except ImportError:
+    pass
+
+if TYPE_CHECKING:
+    from numpy.typing import NDArray
+
+__all__ = ["ValuationResult", "ValuationStatus"]
+
+
+class ValuationStatus(Enum):
+    Pending = "pending"
+    Converged = "converged"
+    MaxIterations = "maximum number of iterations reached"
+    Failed = "failed"
+
+
+@dataclass
+class ValueItem:
+    """The result of a value computation for one datum."""
+
+    index: np.int_
+    name: str
+    value: np.float_
+    stderr: Optional[np.float_]
+
+
+class ValuationResult(collections.abc.Sequence):
+    """Objects of this class hold the results of valuation algorithms.
+
+    Results can be sorted with :meth:`sort`. Note that sorting values affects
+    how iterators and the object itself as ``Sequence`` behave: ``values[0]``
+    returns a :class:`ValueItem` with the highest or lowest ranking point if
+    this object is sorted by descending or ascending value, respectively. If
+    unsorted, ``values[0]`` returns a ``ValueItem`` for index 0.
+
+    Similarly, ``iter(valuation_result)`` returns a generator of
+    :class:`ValueItem` in the order in which the object is sorted.
+
+    :param algorithm: The method used.
+    :param status: The end status of the algorithm.
+    :param values: An array of values, data indices correspond to positions in
+        the array.
+    :param stderr: An optional array of standard errors in the computation of
+        each value.
+    :param data_names: Names for the data points. Defaults to index numbers
+        if not set.
+    :param sort: Whether to sort the values. See above how this affects usage
+        as an iterable or sequence.
+
+    :raise ValueError: If data names and values have mismatching lengths.
+    """
+
+    _indices: "NDArray[np.int_]"
+    _values: "NDArray[np.float_]"
+    _data: Dataset
+    _names: "Union[NDArray[np.int_], NDArray[np.str_]]"
+    _stderr: "NDArray[np.float_]"
+    _algorithm: str  # TODO: BaseValuator
+    _status: ValuationStatus  # TODO: Maybe? BaseValuator.Status
+    _sort_order: Optional[SortOrder] = None
+
+    def __init__(
+        self,
+        algorithm: Callable,  # BaseValuator,
```

Comment:
What is the reason for passing the algorithm function instead of the algorithm name?

Reply:
It's a leftover. It was intended as a placeholder because I was toying with the idea of using objects for the algorithms and having a `BaseValuator`. This would then be passed here and used to query statistics or whatever. But I'm not so sure about that anymore (sklearn interfaces don't fully make sense, only localizing the cache is a good argument), and the callable with no signature is silly

Reply:
Fixed in https://github.com/appliedAI-Initiative/pyDVL/pull/214/commits/2e3a329a35dc051645e10f480b640a94976a3a53